landlab/components/space/tests/test_space.py

```diff
@@ -1,8 +1,234 @@
+        br[mg.core_nodes] = z[mg.core_nodes] - soil[mg.core_nodes]
+    
+    #compare numerical and analytical slope solutions
+    num_slope = mg.at_node['topographic__steepest_slope'][mg.core_nodes]
+    analytical_slope = np.power(U / K_br, 1./n_sp) \
+        * np.power(mg.at_node['drainage_area'][mg.core_nodes], -m_sp / n_sp)
+        
+    #test for match with analytical slope-area relationship
+    testing.assert_array_almost_equal(num_slope, analytical_slope, 
+                                      decimal=8, 
+                                      err_msg='SPACE detachment-limited test failed',
+                                      verbose=True)
+    
+def test_matches_transport_solution():
+    """
+    Test that model matches the transport-limited analytical solution
+    for slope/area relationship at steady state: S=((U * v_s) / (K_sed * A^m) 
+    + U / (K_sed * A^m))^(1/n).
+    """
+    
+    #set up a 3x3 grid with one open outlet node and low initial elevations.
+    nr = 5
+    nc = 5
+    mg = RasterModelGrid((nr, nc), 10.0)
+    
+    z = mg.add_zeros('node', 'topographic__elevation')
+    br = mg.add_zeros('node', 'bedrock__elevation')
+    soil = mg.add_zeros('node', 'soil__depth')
+
+    mg['node']['topographic__elevation'] += mg.node_y / 100000 \
+        + mg.node_x / 100000 \
+        + np.random.rand(len(mg.node_y)) / 10000
+    mg.set_closed_boundaries_at_grid_edges(bottom_is_closed=True,
+                                           left_is_closed=True,
+                                           right_is_closed=True,
+                                           top_is_closed=True)
+    mg.set_watershed_boundary_condition_outlet_id(0,
+                                                  mg['node']['topographic__elevation'], 
+                                                  -9999.)
+    soil[:] += 100. #initial soil depth of 100 m
+    br[:] = z[:]
+    z[:] += soil[:]
+    
+    #Instantiate DepressionFinderAndRouter
+    df = DepressionFinderAndRouter(mg)
+    
+    # Create a D8 flow handler
+    fa = FlowAccumulator(mg, flow_director='D8', 
+                         depression_finder='DepressionFinderAndRouter')
+    
+    # Parameter values for detachment-limited test
+    K_sed = 0.01
+    U = 0.0001
+    dt = 1.0
+    F_f = 1.0 #all detached rock disappears; detachment-ltd end-member
+    m_sp = 0.5
+    n_sp = 1.0
+    v_s = 0.5
+    
+    # Instantiate the Space component...
+    sp = Space(mg, K_sed=K_sed, K_br=0.01,
+                         F_f=F_f, phi=0.0, H_star=1., v_s=v_s,
+                         m_sp=m_sp, n_sp=n_sp, sp_crit_sed=0,
+                         sp_crit_br=0)
+
+    # ... and run it to steady state (4000x1-year timesteps).
+    for i in range(4000):
+        fa.run_one_step()
+        flooded = np.where(df.flood_status==3)[0]
+        sp.run_one_step(dt=dt, flooded_nodes=flooded)
+        br[mg.core_nodes] += U * dt #m
+        soil[0] = 100.
+        z[:] = br[:] + soil[:]
```

Comment:
I don't think I understand why you need to enforce the outlet node soil depth here but not in the detachment case. could you add a comment that states why.